contracts/core/RebalancingSetToken.sol

```diff
@@ -327,9 +331,11 @@ contract RebalancingSetToken is
     function getComponents()
         external
         view
-        returns(address[1])
+        returns(address[])
     {
-        return [currentSet];
+        address[] memory components = new address[](1);
```

Comment:
this seems verbose. with only one value that is already known, you can just initialize the array with it I think: https://ethereum.stackexchange.com/questions/28047/solidity-initialize-memory-array-of-fixed-size

Reply:
try `return new address[currentSet];`

Reply:
The problem is that, that doesn't return a dynamic typed array which wouldn't conform to the getComponents and getUnits interface